identity/app/controllers/EditProfileController.scala

```diff
@@ -156,7 +156,11 @@ class EditProfileController(
 
               case Right(updatedUser) =>
                 logger.info(s"Successfully set hasRepermissioned flag for user ${request.user.id}")
-                SeeOther(returnUrl)
+
+                // Without GU_CONSENTED cookie when users complete re-permissioning journey then they might be again
+                // re-directed to repermissining journey because unsubscribing from V1 newsletters might not have yet
+                // completed as it happens in a fire-and-forget Future in IDAPI
+                SeeOther(returnUrl).withCookies(Cookie("GU_CONSENTED", "true", Some(60)))
```

Comment:
We could convey this information in a returnUrl parameter instead of a cookie? I am not totally against it but people tend to prefer less cookies and some block them (although all sorts would break then)